pkg/api/customization/oci/listers.go

```diff
@@ -0,0 +1,238 @@
+package oci
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/rancher/norman/httperror"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/oracle/oci-go-sdk/common"
+	"github.com/oracle/oci-go-sdk/containerengine"
+	"github.com/oracle/oci-go-sdk/core"
+	"github.com/oracle/oci-go-sdk/identity"
+
+	"encoding/json"
+)
+
+func processVcns(provider common.ConfigurationProvider, compartment string) ([]byte, httperror.ErrorCode, error) {
+	logrus.Debugf("[oci-handler] listing VCNs in compartment: %s", compartment)
+	virtualNetworkClient, err := core.NewVirtualNetworkClientWithConfigurationProvider(provider)
+	if err != nil {
+		logrus.Debugf("[oci-handler] error creating OCI Virtual Network client: %v", err)
+		return nil, httperror.ErrorCode{}, err
```

Comment:
Using an empty struct is going to attempt to use a status of 0, all error returns need a valid code based on what happened.